src/components/SelectionListSingle/BaseSelectionList.tsx

```diff
@@ -0,0 +1,341 @@
+import {useIsFocused} from '@react-navigation/native';
+import {FlashList} from '@shopify/flash-list';
+import type {FlashListRef, ListRenderItem, ListRenderItemInfo} from '@shopify/flash-list';
+import React, {useCallback, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import type {TextInput as RNTextInput, ViewStyle} from 'react-native';
+import {View} from 'react-native';
+import OptionsListSkeletonView from '@components/OptionsListSkeletonView';
+import BaseSelectionListItemRenderer from '@components/SelectionList/BaseSelectionListItemRenderer';
+import ShowMoreButton from '@components/ShowMoreButton';
+import Text from '@components/Text';
+import useArrowKeyFocusManager from '@hooks/useArrowKeyFocusManager';
+import useDebounce from '@hooks/useDebounce';
+import useLocalize from '@hooks/useLocalize';
+import useScrollEnabled from '@hooks/useScrollEnabled';
+import useSingleExecution from '@hooks/useSingleExecution';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CONST from '@src/CONST';
+import areSetsEqual from '@src/utils/setsEqual';
+import Footer from './Footer';
+import SelectionListHeader from './ListHeader';
+import type {DataDetailsType, ListItem, SelectionListProps} from './types';
+
+const ANIMATED_HIGHLIGHT_DURATION =
+    CONST.ANIMATED_HIGHLIGHT_ENTRY_DELAY +
+    CONST.ANIMATED_HIGHLIGHT_ENTRY_DURATION +
+    CONST.ANIMATED_HIGHLIGHT_START_DELAY +
+    CONST.ANIMATED_HIGHLIGHT_START_DURATION +
+    CONST.ANIMATED_HIGHLIGHT_END_DELAY +
+    CONST.ANIMATED_HIGHLIGHT_END_DURATION;
+
+const PAGE_SIZE = CONST.MAX_SELECTION_LIST_PAGE_LENGTH;
+
+function getStartingPage(index: number) {
+    return index >= 0 ? Math.ceil((index + 1) / PAGE_SIZE) : 1;
+}
+
+function BaseSelectionList<TItem extends ListItem>({
+    data,
+    ref,
+    ListItem,
+    textInputOptions,
+    initiallyFocusedItemKey,
+    onSelectRow,
+    onSelectAll,
+    onCheckboxPress,
+    onScrollBeginDrag,
+    onEndReached,
+    onEndReachedThreshold,
+    confirmButtonConfig,
+    aboveListHeaderMessage,
+    customListHeader,
+    footerContent,
+    listEmptyContent,
+    listFooterContent,
+    rightHandSideComponent,
+    alternateNumberOfSupportedLines,
+    selectedItems = [],
```

Comment:
Use a stable array here from CONST.ts: `EMPTY_ARRAY` (without it it is initialized on every render)